### ISSUE TITLE
@Bindable not inherited

### ISSUE THREAD

**Issue by mauromol** [groovy/groovy-eclipse#1083]
Consider this:

```groovy
package test60

import groovy.beans.Bindable
import groovy.transform.CompileStatic

@CompileStatic
class GBean {
  @Bindable
  String foo
  
  void doSomething() {
	  addPropertyChangeListener('foo', { println it.newValue })
  }		
}
```

and this:

```groovy
package test60

import groovy.transform.CompileStatic

@CompileStatic
class Test60 extends GBean {

	void doSomethingElese() {
		addPropertyChangeListener('foo', { println it.newValue })
	}
}
```

The compiler gives a warning when I try to add the property change listener:

    Groovy:[Static type checking] - Cannot find matching method test60.Test60#addPropertyChangeListener(java.lang.String, groovy.lang.Closure). Please check if the declared type is correct and if the method exists.

I'm using Groovy 2.5 compiler. The error seems to suggest a problem with the Groovy compiler (rather than with Greclipse), however I tried to compile this example with Gradle+Groovy (I tested with 2.5.8 and 2.5.11) and it compiles fine.

**Comment by eric-milles** [comment]
I will do a little more looking, but I think this comes down to not processing AST transforms for source references.  If you define the two classes in the same file the transforms will be processed enough to satisfy the source reference.  Or you can define a simple trait that the base class implements just to the method signature is available:
```groovy
trait Binder {
  void addPropertyChangeListener(String name, PropertyChangeListener listener) {
  }
}
class GBean implements Binder {
  ...
}
```

**Comment by mauromol** [comment]
There's a further inconsistency though.

I mean, in addition to the steps to repro above, try to add a `@Bindable` property to `Test60` class as well, like this:

```groovy
package test60

import groovy.beans.Bindable
import groovy.transform.CompileStatic

@CompileStatic
class Test60 extends GBean {

	@Bindable
	String bar
	
	void doSomethingElese() {
		addPropertyChangeListener('foo', { println it.newValue })
	}
}
```
Now, in some way the transform should be defined (also) in the same source file. Indeed, the call to `addPropertyChangeListener` is shown in a different colour (it seems like syntax highlighting recognises it as a method correctly), however the red underlining and the error mark on the vertical ruler are still saying that there's no `addPropertyChangeListener` method.

On another project of mine, instead, I have another different behaviour. The case is similar to the latter one (so a `@Bindable` property exists in the subclass as well), but not only `addPropertyChangeListener` method is not found and not coloured correctly, it's even shown with the "black" underlining (in addition to the red one) that is typical of dynamic Groovy (where a method is not resolved but not rejected due to the dynamic type checking), even though the class is `@CompileStatic`...

**Comment by eric-milles** [comment]
If you have:
```groovy
class C {
  @groovy.beans.Bindable Type prop
}
```
Your class will gain `addPropertyChangeListener` and some other methods that `Bindable` provides.  You should be able to use those additions within `C` itself (and probably the unit that contains `C`) and there should be no compiler errors or warning markers.

![image](https://user-images.githubusercontent.com/18193802/79358750-3f329680-7f07-11ea-8f72-4656d519a275.png)

Content assist for the new methods is provided through the Groovy DSL Support container.  If you don't have it on your project or DSLD is disabled, you will not get some extra assistance in the editor.

Once you move to another source, references to `C` are not processed as completely (for efficiency) and so you are missing out on AST transform additions.  This is the same for `@Canonical`, `@Delegate`, etc.  Issue #704 and others speak to this deficiency.

**Comment by eric-milles** [comment]
With the addition of a Java interface with a default method, I am able to work with a subclass in a different source unit with no errors:
![image](https://user-images.githubusercontent.com/18193802/79359479-47d79c80-7f08-11ea-8a36-5d71d77fdc1f.png)

**Comment by mauromol** [comment]
OK, but why do you think my latter case is not working? I.e. Why in `Test60` (having itself another `@Bindable` property) `addPropertyChangeListener` is not found by the compiler but seems to be found by syntax highlighting?

**Comment by eric-milles** [comment]
I am not getting any compiler errors.  If you replace `{ println it.newValue }` with `{ println it.newValue } as PropertyChangeListener`, does it make a difference?

**Comment by mauromol** [comment]
No:

![immagine](https://user-images.githubusercontent.com/1742494/79367398-21d4eb00-7f4e-11ea-821c-85434e62d05c.png)

As you can see, `addPropertyChangeListener` is green, but there's the compiler error.

**Comment by mauromol** [comment]
If I comment out the local `@Bindable` property, the compiler error is the same, but the `addPropertyChangeListener` call turns white:

![immagine](https://user-images.githubusercontent.com/1742494/79367572-652f5980-7f4e-11ea-93a8-40e59c65f381.png)

**Comment by eric-milles** [comment]
Is your type GBean implementing a Java interface like this (suggested workaround)?
```java
interface Binder {
  default void addPropertyChangeListener(String name, java.beans.PropertyChangeListener listener) {}
}
```

**Comment by mauromol** [comment]
No, no, I'm not applying any workaround. I know that implementing/extending something that provide those methods works, because I already did it in other circumstances.
Just wanted to show you about a further inconsistency apart from the fact that AST are not processed across different source files. Indeed, in the case where `@Bindable` is also present in `Test60` I would have expected it to work, at least...

**Comment by eric-milles** [comment]
Okay, so if you don't make the method well known to referencing classes, you will fall back into the existing issue space.  AST transforms are not applied when source references are processed.  There are other open tickets for the root cause.

**Comment by mauromol** [comment]
I don't know whether the fix for #421 is supposed to impact on this and/or on #704.
In any case, I still see this problem in Greclipse 4.8.
